core/src/main/java/org/axonframework/commandhandling/model/AggregateLifecycle.java

```diff
@@ -120,6 +129,13 @@ protected static AggregateLifecycle getInstance() {
      */
     protected abstract boolean getIsLive();
 
+    /**
+     * Gets the version of the aggregate.
+     *
+     * @return the current version of the aggregate
+     */
+    protected abstract Long doGetVersion();
```

Comment:
Is it possible to "promote" the existing `version()` method to this class, instead? I don't feel this method adds a lot to the API.